:farmer: Rci__nightly-release 3 clang and cmake warnings by an `uninitialized static_peer`

### Generated by Generative AI

_No response_

### Operating System:

Ubuntu 24.04

### ROS version or commit hash:

Rolling

### RMW implementation (if applicable):

rmw_cyclonedds_cpp

### RMW Configuration (if applicable):

_No response_

### Client library (if applicable):

_No response_

### 'ros2 doctor --report' output

_No response_

### Steps to reproduce issue

1. Go to http://build.ros2.org/job/Rci__nightly-release_ubuntu_noble_amd64
2. Build a Job
3. See the warning

### Expected behavior

To build without warnings

### Actual behavior

Building with warnings

### Additional information

### Reference Build
- [Rci__nightly-release_ubuntu_noble_amd64#696](http://build.ros2.org/job/Rci__nightly-release_ubuntu_noble_amd64/696)

### Log Output
`rmw_node.cpp:1184`:
```
07:02:31     inlined from ‘rmw_ret_t rmw_context_impl_s::init(rmw_init_options_t*, std::size_t)’ at /tmp/ws/src/ros2/rmw_cyclonedds/rmw_cyclonedds_cpp/src/rmw_node.cpp:1380:27:
07:02:31 /tmp/ws/src/ros2/rmw_cyclonedds/rmw_cyclonedds_cpp/src/rmw_node.cpp:1184:36: warning: ‘add_localhost_as_static_peer’ may be used uninitialized [-Wmaybe-uninitialized]
07:02:31  1184 |       add_localhost_as_static_peer ||
07:02:31       |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
07:02:31  1185 |       add_static_peers ||
07:02:31       |       ~~~~~~~~~~~~~~~~              
07:02:31 /tmp/ws/src/ros2/rmw_cyclonedds/rmw_cyclonedds_cpp/src/rmw_node.cpp: In member function ‘rmw_ret_t rmw_context_impl_s::init(rmw_init_options_t*, std::size_t)’:
07:02:31 /tmp/ws/src/ros2/rmw_cyclonedds/rmw_cyclonedds_cpp/src/rmw_node.cpp:1134:10: note: ‘add_localhost_as_static_peer’ was declared here
07:02:31  1134 |     bool add_localhost_as_static_peer;
07:02:31       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Apparently this started when rolling builds were down, first appearance is https://build.ros2.org/job/Rci__nightly-release_ubuntu_noble_amd64/675/

CC: @Crola1702

--- a/rmw_cyclonedds_cpp/src/rmw_node.cpp
+++ b/rmw_cyclonedds_cpp/src/rmw_node.cpp
@@ -1131,9 +1131,9 @@ static bool check_create_domain(dds_domainid_t did, rmw_discovery_options_t * di
     dom.refcount = 1;
     dom.discovery_options = *discovery_options;
 
-    bool add_localhost_as_static_peer;
-    bool add_static_peers;
-    bool disable_multicast;
+    bool add_localhost_as_static_peer = false;
+    bool add_static_peers = false;
+    bool disable_multicast = false;
 
     switch (discovery_options->automatic_discovery_range) {
       case RMW_AUTOMATIC_DISCOVERY_RANGE_NOT_SET:

